jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java

```diff
@@ -62,8 +62,6 @@
     public static final String PESSIMISTIC_WRITE_LOCK_MODE = "pessimistic_write";
     public static final String NONE_LOCK_MODE = "none";
 
-    private final transient boolean isTraceEnabled = ROOT_LOGGER.isTraceEnabled();
```

Comment:
It's an instance variable, and I believe, but could be wrong, that these are per-request objects. If so, then the current code is a reasonable perf optimization by removing possibly numerous calls to isTraceEnabled(). The only downside is the inability to react to a logging config change in the middle of a request.